content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx

```diff
@@ -127,4 +127,109 @@ userPosts.posts
+
+The first tab below shows an example Prisma schema with a one-to-many relation between `User` and `Post`. The second tab shows the `PostUncheckedCreateInput` generated type. It contains the `authorId` property, which is a foreign key. The third tab shows an example query that uses the `PostUncheckedCreateInput` type. This query will result in an error if a user with an `id` of `1` does not exist.
+
+<TabbedContent tabs={[<FileWithIcon text="Model" icon="prisma"/>, <FileWithIcon text="Generated type" icon="code"/>, <FileWithIcon text="Example query" icon="code"/>]}>
+
+<tab>
+
+```prisma
+model Post {
+  id       Int     @id @default(autoincrement())
+  title    String  @db.VarChar(255)
+  content  String?
+  author   User    @relation(fields: [authorId], references: [id])
+  authorId Int
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+  posts Post[]
+}
+```
+
+</tab>
+
+<tab>
+
+```ts
+type PostUncheckedCreateInput = {
+  id?: number
+  title: string
+  content?: string | null
+  authorId: number
+}
+```
+
+</tab>
+<tab>
+
+```ts
+prisma.post.create({
+  data: {
+    title: 'First post',
+    content: 'Welcome to the first post in my blog...',
+    authorId: 1,
+  },
+})
+```
+
+</tab>
+
+</TabbedContent>
+
+The same query can be rewritten using the "safer" `PostCreateInput` type. This type does not contain the `authorId` field but instead contains the `author` relation field.
+
+<TabbedContent tabs={[ <FileWithIcon text="Generated type" icon="code"/>, <FileWithIcon text="Example query" icon="code"/>]}>
+
+<tab>
+
+```ts
+type PostCreateInput = {
+  title: string
+  content?: string | null
+  author: UserCreateNestedOneWithoutPostsInput
+}
+
+type UserCreateNestedOneWithoutPostsInput = {
+  create?: XOR<
+    UserCreateWithoutPostsInput,
+    UserUncheckedCreateWithoutPostsInput
+  >
+  connectOrCreate?: UserCreateOrConnectWithoutPostsInput
+  connect?: UserWhereUniqueInput
+}
+```
+
+</tab>
+<tab>
+
+```ts
+prisma.post.create({
+  data: {
+    title: 'First post',
+    content: 'Welcome to the first post in my blog...',
+    author: {
+      connect: {
+        id: 1,
+      },
+    },
+  },
+})
+```
+
+</tab>
+
+</TabbedContent>
+
+This query will also result in an error if an author with an `id` of `1` does not exist. In this case, Prisma Client will give a more descriptive error message. You can also use the [`connectOrCreate`](../../../../reference/api-reference/prisma-client-reference#connectorcreate) API to safely create a new user if one does not already exist with the given `id`.
+
+We recommend using the "safe" `Input` types whenever possible. However, there are some cases where you may need to use the "unsafe" `UncheckedInput` types.
```

Comment:
What are those use cases?

Reply:
This is incorrectly phrased. It's not as much an alternative use case as it is for an alternative syntax. I'll correct it. Thanks for picking this up